pubmed_dl/main.py

```diff
@@ -1,38 +1,53 @@
-import json
-from pubmed_dl.ncbi import get_list_pmid, uids_to_docs
-import logging
-import os
-from pathlib import Path
-from dotenv import load_dotenv
-from pydantic import BaseSettings
-import time
-
-
-dot_env_filepath = Path(__file__).absolute().parent.parent / ".env"
-load_dotenv(dot_env_filepath)
-
-
-class Settings(BaseSettings):
-    loglevel: str = os.getenv("LOGLEVEL", "INFO")
-
-
-settings = Settings()
-logging.basicConfig(level=settings.loglevel)
-
-
-def main():
-    start_time = time.time()
-    start = input("Enter start date (YYYY/MM/DD): ")
-    end = input("Enter end date (YYYY/MM/DD): ")
-    val = get_list_pmid(start, end)
-    with open("test.json", "w") as f:
-        for batch in uids_to_docs(val):
-            for doc in batch:
-                f.write(f"{json.dumps(doc)}\n")
-    logging.info(f"Done writing data from {start} to {end} onto file named: test.json")
-    duration = time.time() - start_time
-    logging.info(f"Total run time for {len(val)} docs: {duration}s")
-
-
-if __name__ == "__main__":
-    main()
+import json
+from pubmed_dl.ncbi import get_list_pmid, uids_to_docs
+import logging
+import os
+from pathlib import Path
+from dotenv import load_dotenv
+from pydantic import BaseSettings
+import time
+import typer
+
+
+dot_env_filepath = Path(__file__).absolute().parent.parent / ".env"
+load_dotenv(dot_env_filepath)
+
+
+class Settings(BaseSettings):
+    loglevel: str = os.getenv("LOGLEVEL", "INFO")
+
+
+settings = Settings()
+logging.basicConfig(level=settings.loglevel)
+
+
+def main(start: str, end: str, request_file: str):
+    """
+    Main requires 3 arguments. 
+    
+    The first one is start date (eg: YYYY/MM/DD).
+
+    The second one is end date (eg: 2021/02/05).
+    
+    The third one is the filename where the data needs to be written to.
+
+    """
+    start_time = time.time()
+    output_filepath: Path = Path(request_file)
+    output_filepath.parents[0].mkdir(parents=True, exist_ok=True)
+    filename = request_file
```

Comment:
Why create a third variable? Just use `output_filepath` henceforth (e.g. in the call to `with open`)